frontend/src/components/App.js

```diff
@@ -1,26 +1,79 @@
 import React from 'react';
 import EventList from "./EventList";
 import './App.css';
+import Grid from "@material-ui/core/Grid";
 
-function App() {
-    let events = [
-        {
-            "description": "Room 312",
-            "start": "2020-07-25",
-            "end": "2020-07-26",
-        },
-        {
-            "description": "Room 311",
-            "start": "2020-07-27",
-            "end": "2020-07-29",
+import Paper from "@material-ui/core/Paper";
+import DatePicker from "./DatePicker";
+
+class App extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            events: [],
+            start: new Date()
+        }
+    }
+
+    componentDidMount() {
+        fetch("http://localhost:5000/api/calendar/", {
+            method: 'POST',
+            body: JSON.stringify(this.state.start)
+        }) //todo move to prop file
+            .then(res => res.json())
+            .then(json => {
+                this.setState({events: json})
+            });
+    }
+
+    async componentDidUpdate(prevProps, prevState) {
+        if (this.state.start !== prevState.start) {
```

Comment:
Async await here in unnecessary.

Also, it would be greater to use AbortController to cancel previous requests if there any.
https://developer.mozilla.org/ru/docs/Web/API/AbortController
https://robwise.github.io/blog/cancel-whatwg-fetch-requests-in-react

Reply:
Fixed.